src/multiselect.js

```diff
@@ -230,26 +232,75 @@ export class Multiselect {
+              preCondition(selected)) {
             if (ws.doubleClickPid_) {
               clearTimeout(ws.doubleClickPid_);
               ws.doubleClickPid_ = undefined;
-              if (Blockly.getSelected().id === ws.doubleClickBlock_) {
-                const state = !Blockly.getSelected().isCollapsed();
+              if (selected.id === ws.doubleClickBlock_) {
+                const state = !selected.isCollapsed();
                 const maybeCollapse = function(block) {
                   if (block && preCondition(block) &&
-                  !hasSelectedParent(block)) {
+                      !hasSelectedParent(block)) {
                     block.setCollapsed(state);
                   }
                 };
                 Blockly.Events.setGroup(true);
-                const blockSelection = blockSelectionWeakMap.get(ws);
-                if (Blockly.getSelected() && !blockSelection.size) {
-                  maybeCollapse(Blockly.getSelected());
+                if (selected) {
+                  maybeCollapse(selected);
+                }
+                Blockly.Events.setGroup(false);
+                return;
+              }
+            }
+            if (!ws.doubleClickPid_) {
+              ws.doubleClickBlock_ = selected.id;
+              ws.doubleClickPid_ = setTimeout(function() {
+                ws.doubleClickPid_ = undefined;
+              }, 500);
+            }
+          } else if (selected && selected instanceof MultiselectDraggable) {
+            // Case where the selected is a multidraggable instance
+            if (ws.doubleClickPid_) {
+              clearTimeout(ws.doubleClickPid_);
+              ws.doubleClickPid_ = undefined;
+              const dragSelection = dragSelectionWeakMap.get(ws);
+              if (dragSelection.size) {
+                // Checking whether any of the blocks in
+                // the dragSelection is not collapsed.
+                // If there are not collapsed blocks,
+                // set the maybeCollapse function to collapse
+                // those uncollapsed blocks.
+                // Otherwise, uncollapse all the collapsed blocks.
+                let notCollapsed = 0;
+                dragSelection.forEach((id) => {
+                  if (ws.getBlockById(id)) {
+                    if (!ws.getBlockById(id).isCollapsed() &&
+                        !hasSelectedParent(ws.getBlockById(id))) {
+                      notCollapsed += 1;
+                    }
+                  }
+                });
+                let state = false;
+                if (notCollapsed > 0) {
+                  state = true;
                 }
-                blockSelection.forEach(function(id) {
+
+                const maybeCollapse = function(block) {
+                  if (block && preCondition(block) &&
+                      !hasSelectedParent(block)) {
+                    block.setCollapsed(state);
+                  }
+                };
```

Comment:
Let's move the maybeCollapse function one level up, above the if branch, to avoid duplication of code.